buffer/src/main/java/io/netty5/buffer/api/DefaultCompositeBuffer.java

```diff
@@ -927,6 +927,12 @@ public int countWritableComponents() {
     @Override
     public <E extends Exception> int forEachReadable(int initialIndex, ReadableComponentProcessor<E> processor)
             throws E {
+        if (!isAccessible()) {
+            throw attachTrace(bufferIsClosed(this));
+        }
+        if (readableBytes() == 0) {
+            return 0;
+        }
         checkReadBounds(readerOffset(), Math.max(1, readableBytes()));
```

Comment:
We don't have to do these `Math.max(1, …)` calls anymore, because now the readable/writable bytes are always greater than zero.

Reply:
Fixed... PTAL again